skyreader/utils/handle_map_data.py

```diff
@@ -101,14 +112,21 @@ def extract_map(
         # show 2 * delta_LLH
         grid_value = grid_value * 2.
         equatorial_map *= 2.
+
+        # nan values are a problem for the convolution and the contours
+        max_map = np.nanmax(equatorial_map)
+        equatorial_map[np.isnan(equatorial_map)] = max_map
+        grid_value[np.isnan(grid_value)] = max_map
+        grid_value = grid_value.clip(None, max_map)
     else:
         # Convert to probability
         equatorial_map = np.exp(-1. * equatorial_map)
         equatorial_map = equatorial_map / np.nansum(equatorial_map)
 
         # nan values are a problem for the convolution and the contours
-        min_map = np.nanmin(equatorial_map)
+        min_map = np.nanmin(equatorial_map[equatorial_map > 0.])
         equatorial_map[np.isnan(equatorial_map)] = min_map
+        equatorial_map[equatorial_map == 0.] = min_map
```

Comment:
I do it here because of the line `equatorial_map = np.exp(-1. * equatorial_map)` (line 122). Using float64, values smaller than around 10-300 are set equal to zero, therefore some zeros might pop out at this point and not before. I briefly thought about using float128 but I think it might me hard as many functions do not support this type of data and it is probably not worth for keeping so small probabilities.

Reply:
Yes taking `exp(-x)` for large x will hit 0, but why not just keep the 0s? This was the behavior before the changes in the PR.

Reply:
In the plot of the all-sky map we show the the log10 of the probability, and also to calculate contours we use the logarithm. This was the reason why I added that, but we can also keep the 0s. For this purpose I now added some logic to handle them in the all-sky map and in the contour calculation. Otherwise, the 0s are always kept.